### PR TITLE
add password policy migration

### PR DESCRIPTION
**Pull request by brian-mw**
### Enforce password policies using migration

- Update password_reset_interval to 30 days
- Enable Password Policies for all
- Force users to immediately reset password

### Add password validation

- Length Must be >= 6 Characters

### CODE REVIEW

File: db/migrate/20241011080330_password_policies.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    user_props = UserProperty
+
+    # enable password policy
+    props.find_by_property('password_policy_enabled')&.update(property_value: 'true')
+
+    # set password reset interval to 30 days
+    props.find_by_property('password_reset_interval')&.update(property_value: '30')
+
+    # set all user props to forcefully reset passwords
+    user_props.where(property: 'last_password_reset').update_all(property_value: Date.today - 31.days)
```
Comment (Skipper-116) [inline db/migrate/20241011080330_password_policies.rb]: We should also add logic to add last_pasword_reset for users without it.

**Review by Skipper-116 — CHANGES_REQUESTED** [review]
Lets handle sites having users without the last_password_change property as well

File: db/migrate/20241011080330_password_policies.rb (hunk 1 above)
Comment (brian-mw) [reply db/migrate/20241011080330_password_policies.rb]: Updated

**Review by Skipper-116 — APPROVED** [review]

**Review by Skipper-116 — CHANGES_REQUESTED** [review]
Add functionality to store last six passwords to avoid reuse of old passwords

**Comment by brian-mw** [comment]
> Add functionality to store last six passwords to avoid reuse of old passwords

**Review by KytonThaundi — APPROVED** [review]

**Review by Skipper-116 — APPROVED** [review]

**Review by KytonThaundi — APPROVED** [review]

**Pull request merged**